test/agent/Agent.js

```diff
@@ -0,0 +1,67 @@
+const agent = require('superagent');
+
+class Agent {
```

Comment:
Agent is a weird name for that class and it's confusing with SuperAgent, avoid using the same names that your imported libraries because you are creating dependencies. What happens if you use another library and it doesn't have the Agent concept?